pkg/kademlia/kademlia.go

```diff
@@ -275,16 +277,23 @@ func (k *Kademlia) FetchInfo(ctx context.Context, node pb.Node) (_ *pb.InfoRespo
 	return info, nil
 }
 
-// FindNode looks up the provided NodeID first in the local Node, and if it is not found
-// begins searching the network for the NodeID. Returns and error if node was not found
+// FindNode wraps a call to FindNodeWithVouchers with no vouchers
 func (k *Kademlia) FindNode(ctx context.Context, nodeID storj.NodeID) (_ pb.Node, err error) {
+	defer mon.Task()(&ctx)(&err)
+	return k.FindNodeWithVouchers(ctx, nodeID, nil, nil)
+}
+
+// FindNodeWithVouchers looks up the provided NodeID first in the local Node, and if it is not found
+// begins searching the network for the NodeID. When it contacts nodes, it will send vouchers to them
+// and request to be put into their routing table. Returns and error if node was not found
+func (k *Kademlia) FindNodeWithVouchers(ctx context.Context, nodeID storj.NodeID, vrs []*pb.Voucher, self *pb.Node) (_ pb.Node, err error) {
 	defer mon.Task()(&ctx)(&err)
 	if !k.lookups.Start() {
 		return pb.Node{}, context.Canceled
 	}
 	defer k.lookups.Done()
 
-	results, err := k.lookup(ctx, nodeID)
+	results, err := k.lookup(ctx, nodeID, vrs, self)
```

Comment:
nodeID of self , can be obtained selfnode := self.Local().Node, then why send self as an argument. What do you think?

Reply:
sometimes when we call lookup, we want to pass self, other times we don't
related: https://github.com/storj/storj/pull/2652#discussion_r308390951